viz_scripts/timeseries_analysis.ipynb

```diff
@@ -0,0 +1,471 @@
+    "ax.figure.savefig(\"/plots/ts_ebike_share%s.png\" % file_suffix)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of total ebikeshare mileage across all users\n",
+    "plot_data = data[data['Mode_confirm']=='Pilot ebike']\n",
+    "plot_data = plot_data.groupby(['date_time'], as_index=False)['distance_miles'].agg(['sum'])\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(16,4))\n",
+    "sns.lineplot(ax=ax, data=plot_data, x='date_time', y='sum').set(title='Daily E-Bike Mileage', xlabel='Date', ylabel='Miles')\n",
+    "plt.xticks(rotation=45)\n",
+    "plt.subplots_adjust(bottom=0.25)\n",
+    "ax.figure.savefig(\"/plots/ts_ebike_miles%s.png\" % file_suffix)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of total ebikeshare mileage normalized by number of users\n",
+    "plot_data = data[data['Mode_confirm']=='Pilot ebike']\n",
+    "plot_data = plot_data.groupby(['date_time'], as_index=False)['distance_miles'].agg(['sum'])\n",
+    "plot_data = plot_data.merge(active_users, on='date_time')\n",
+    "plot_data['mileage_per_user'] = plot_data['sum'] / plot_data['active_users']\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(16,4))\n",
+    "sns.lineplot(ax=ax, data=plot_data, x='date_time', y='mileage_per_user').set(title='Daily E-Bike Mileage per Active User', xlabel='Date', ylabel='miles/user')\n",
+    "plt.xticks(rotation=45)\n",
+    "plt.subplots_adjust(bottom=0.25)\n",
+    "ax.figure.savefig(\"/plots/ts_ebike_miles_user%s.png\" % file_suffix)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of mode share proportions across all users\n",
+    "plot_data = mode_counts.groupby(['date_time','Mode_confirm'], as_index=False)['trip_count_smooth'].sum()\n",
+    "total_trips = plot_data.groupby(['date_time'], as_index=False).sum()\n",
+    "plot_data = plot_data.merge(total_trips, on='date_time')\n",
+    "plot_data['trip_proportion'] = plot_data['trip_count_smooth_x'] / plot_data['trip_count_smooth_y']\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(16,4))\n",
+    "sns.lineplot(ax=ax, data=plot_data, x='date_time', y='trip_proportion', hue='Mode_confirm').set(title='Daily Aggregate Mode Share', xlabel='Date', ylabel='Proportion of All Trips')\n",
+    "plt.xticks(rotation=45)\n",
+    "plt.subplots_adjust(bottom=0.25)\n",
+    "plt.legend(loc='center left', title='Confirmed Mode')\n",
+    "ax.figure.savefig(\"/plots/ts_all_modes%s.png\" % file_suffix)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of ebike mode share across many individual users\n",
+    "# This plot is a little crazy if the number of users displayed is not limited\n",
```

Comment:
Yeah that is fine, most programs have < 50 users each.
Can you add an example of what this looks like for the first 50 users, for example?
we can just leave it off the metrics list if it looks too wonky.

Reply:
Here is an example with the first 20 user IDs:

![ts_ebike_mile_individual_prepilot](https://user-images.githubusercontent.com/5828311/178074088-70a4fab9-d35b-4e82-9483-ac5328735a6e.png)

Reply:
hm that is not as pretty as I had hoped. I wonder if a monthly bar graph would be more readable than a daily graph.
I will go ahead and publish this now, but if you are looking for something to do, trying to split by month and/or show as a bar graph may be more readable.

Reply:
or even weekly

And maybe a second one for trip count instead of distance.